File: hello-typescript/src/client/core/ChunkManager.ts
/**
 * Chunk manager for handling data chunking operations.
 * Provides utilities for chunk size calculations and data splitting.
 * Matches Java ChunkManager functionality.
 */

export class ChunkManager {
  private static readonly DEFAULT_CHUNK_SIZE = 65536; // 64KB
  private static readonly UPLOAD_CHUNK_SIZE = 8192; // 8KB for upload to avoid WebSocket fragmentation

  getDefaultChunkSize(): number {
    return ChunkManager.DEFAULT_CHUNK_SIZE;
  }

  getUploadChunkSize(): number {
    return ChunkManager.UPLOAD_CHUNK_SIZE;
  }

  calculateChunkCount(fileSize: number, chunkSize: number): number {
    return Math.ceil(fileSize / chunkSize);
  }

  getChunkSize(
    fileSize: number,
    offset: number,
    defaultChunkSize: number,
  ): number {
    const remainingBytes = fileSize - offset;
    return Math.min(defaultChunkSize, remainingBytes);
  }
}

File: hello-python/src/client/core/chunk_manager.py
"""Chunk manager for handling file chunking operations"""

CHUNK_SIZE = 65536  # 64KB
UPLOAD_CHUNK_SIZE = 8192  # 8KB for WebSocket to avoid fragmentation


class ChunkManager:
    """Manager for chunk operations"""
    
    @staticmethod
    def get_chunk_size() -> int:
        """Get default chunk size for file operations"""
        return CHUNK_SIZE
    
    @staticmethod
    def get_upload_chunk_size() -> int:
        """Get chunk size for upload operations"""
        return UPLOAD_CHUNK_SIZE
    
    @staticmethod
    def calculate_chunks_needed(file_size: int, chunk_size: int) -> int:
        """Calculate number of chunks needed for a file"""
        return (file_size + chunk_size - 1) // chunk_size
